### DISCUSSION TITLE
Unwrap Large Images

### DISCUSSION THREAD

**Discussion by krkrueger** [gmtsar/gmtsar#843] [Q&A]
I had a question about unwrapping large images. There has been some interest in looking at really large areas of which snaphu does not like to solve all at once.  I have been able to successfully unwrap images of pixel size 5000X22000 without too much issue.  But I seem to have found that 12000x35000 is too much for the system.

I found an option in SNAPHU that allows for tiling.  So I figured to give that a try, if 5000x22000 worked, then I though maybe do a 3x3 tile, with ignoring overlap for simple math should have dropped the single tile sizes down to 4000x11000 which should be in the range of what had worked before.  That did not seem to work, so kept increasing the number of tiles down to 10x10 still failing.  Although the error changed to, too many secondary arcs, increase minregion parameter, or decrease some threshold parameter. I feel like I am just randomly increasing/decreasing things, running unwrapping for awhile and then seeing errors and starting over.  I was hoping for something a little more systematic.

So long story short, what is the best way you guys can recommend to do the unwrapping over a large area, that would then be able to be used in a time series?

**Comment by Xiaohua-Eric-Xu** [comment]
I my self never tested the tile mode because, what SNAPHU solves is the global minimum length among residues (like 0 norm). Solving part of the image and piece together may grant you a different solution (not tested). There is a upper limit of lines in SNAPHU which we met before unwrapping high-resolution ALOS-2 ScanSAR data. Other than that, the memory cost is not small either when you have a large image like that. 

I would always ask myself whether I need so high resolution of an image, since filtering and downsampling will help solve the problem. If you need a high-resolution large-sacle image unwrapped, you may also try some other unwrappers, or even use some sparse unwrappers for PS analysis such as StaMPs.

p.s., we recently update the SNAPHU version to 2.0 but I am not sure whether that helps with the problem.

**Reply by krkrueger** [reply]
Thanks for the response. Unfortunately, the desired result needs to be as high of resolution as possible and can't be cut down. So maybe other unwrappers are the solution,  tiling seems to work okay in some places, other places it seems to give very different answers.  I think it comes down to the same effect that you get when you unwrap subswaths before you merge vs after you merge.  The larger area gives more opportunity to avoid the unwrapping errors it seems.

**Reply by Xiaohua-Eric-Xu** [reply]
Another way might be to 
1) unwrap a downsampled interferogram
2) upsample the interferogram to the resolution of your original interferogram
3) difference the two and unwrap the residual interferogram in tile mode (this'll reduce errors I think)
4) add back the unwrapped interferogram in tile mode, to the upsampled interferogram. 

Let me know if this works.

**Reply by krkrueger** [reply]
I will give this a test, looks interesting.  I have been running some of my subsequent intfs that are small enough to do single, both with and without tiling, and at least so far haven't noticed anything egregious yet.  But I will give this method a whirl as well and report back the findings. Basically, you are looking at doing the unwrapping for low frequencies, and then high frequencies.

**Reply by Xiaohua-Eric-Xu** [reply]
> Basically, you are looking at doing the unwrapping for low frequencies, and then high frequencies.

Bingo, exactly!
